lib/rules/order-in-components.js

```diff
@@ -1,60 +1,72 @@
 'use strict';
 
-var utils = require('../utils/utils');
 var ember = require('../utils/ember');
+var reportUnorderedProperties = require('../utils/property-order').reportUnorderedProperties;
+
+const ORDER = [
+  'service',
+  'property',
+  'single-line-function',
+  'multi-line-function',
+  'observer',
+  'lifecycle-hook',
+  'actions',
+  'method',
+  'unknown',
+];
+
+const NAMES = {
+  'service': 'service injection',
+  'property': 'property',
+  'single-line-function': 'single-line function',
+  'multi-line-function': 'multi-line function',
+  'observer': 'observer',
+  'lifecycle-hook': 'lifecycle hook',
+  'actions': 'actions hash',
+  'method': 'custom method',
+  'unknown': 'unknown property type',
+};
+
+function toType(node) {
+  if (ember.isInjectedServiceProp(node.value)) {
+    return 'service';
+  } else if (ember.isCustomProp(node)) {
+    return 'property';
+  } else if (ember.isSingleLineFn(node)) {
+    return 'single-line-function';
+  } else if (ember.isMultiLineFn(node)) {
+    return 'multi-line-function';
+  } else if (ember.isObserverProp(node.value)) {
+    return 'observer';
+  } else if (ember.isComponentLifecycleHook(node)) {
+    return 'lifecycle-hook';
+  } else if (ember.isActionsProp(node)) {
+    return 'actions';
+  } else if (ember.isComponentCustomFunction(node)) {
+    return 'method';
+  } else {
```

Comment:
This `else` is redundant - if we got that far, we can just `return` directly. I understand it's a matter of style preference, though.